More Cleanup + Final Content for 5.x.x

Witchybun opened the pull request:
Includes the following:
- Hat Mouse Lacey support
  - Includes new region and quests associated with her
- Boarding House and Bus Stop Extension support
  - Adds new regions, quests, recipes, crafts, as well as some new locations based on the chests.
  - Adds monster location support for the underground areas
- Makes Magic Elixir a cooking location, not crafting (oops!)
- Add support for Aurora Vineyard bundle pricing (essentially, ship bundle price setting to client)

This *should* be the final major PR regarding 5.x.x, so also let me know if you believe something has yet to be included.

agilbert1412 commented on worlds/stardew_valley/data/locations.csv:
```diff
@@ -2567,6 +2626,7 @@ id,region,name,tags,mod_name
 7523,Boarding House - First Floor,Pumpkin Soup,"MANDATORY,STORY_QUEST",Boarding House and Bus Stop Extension
 7524,Museum,Geode Order,SPECIAL_ORDER_BOARD,Professor Jasper Thomas
 7525,Museum,Dwarven Scrolls,SPECIAL_ORDER_BOARD,Professor Jasper Thomas
+7526,Mouse House,Hats for the Hat Mouse,"MANDATORY,STORY_QUEST",Hat Mouse Lacey
```
Story quests should not be mandatory

Witchybun replied:
Yeah I'm realizing that.  Going through a lot of old code and properly linking quests and rewards and realizing the pain that is non-mandatory quests. I cry.

agilbert1412 commented on worlds/stardew_valley/rules.py:
```diff
@@ -948,3 +951,16 @@ def set_sve_ginger_island_rules(logic: StardewLogic, multiworld: MultiWorld, pla
                              logic.tool.has_tool(Tool.pickaxe, ToolMaterial.iron) & logic.tool.has_tool(Tool.axe, ToolMaterial.iron))
+
+
+def set_boarding_house_rules(logic: StardewLogic, multiworld: MultiWorld, player: int, world_options: StardewValleyOptions):
```
This isn't called?

Witchybun replied:
Fixed it so its called, was an oversight.  Thanks!

Witchybun commented:
For a second glance over I'll note a few changes I just did since it might not be obvious:
- Removed Distant Land seed items.  It turns out that doing the quest regarding the crops lets them be foraged from the swamp map.  I try to keep the spirit of those mods in terms of how they're acquired the same in some way, so right now the logic is if the quest is completed so the crops show up.
- Removed Lance's Wand Chest location.  Its was a double dip location (quest, whose reward is the chest, and then the chest).  The client will now send the Monster Crops location when you open the chest.
- Three quests were cut down as they were double dipping in terms of recipes: Witch's Order/Corrupted Crops Task (essentially one quest depending on Alecto), Pumpkin Soup, and A New Pot.  So the Crayfish Soup, Special Pumpkin Soup, and Ginger Tincture Recipe locations are removed (as they are quest rewards).  Client will handle whether or not to give these out if quests are disabled.
- Fixed the rules for Boarding House entrances but cut a lot of it out.  Interestingly even though there are boulders , branches, and stones here, all of them can be cleared out with a basic tool.  So right now no real rules for tools are needed despite boulders usually needing iron.
- After testing before, erroneously thought the mines were clear enough, but I filled out the monster locations for everything so its more clear now.  Abandoned Mines basically have monsters you'd expect in mines 1~30.  There's Pterodactyls in the Lost Valley but they are just reskinned iridium bats.

agilbert1412 reviewed (changes requested):
The csvs are broken again
